src/rccl_wrap.cc

```diff
@@ -477,23 +477,26 @@ void rcclSetWarpSpeedAuto(struct ncclComm* comm, struct ncclTaskColl* info, size
   if(!comm->topo->warpSpeedEnabled) {
     return;
   }
- info->useWarpSpeed = (info->algorithm == NCCL_ALGO_RING); // Enabled by default for any RING algorithm when platform supports it
-  if(rcclParamWarpSpeedAutoMode() != 0 && IsArchMatch(comm->archName, "gfx950")) { // Auto mode only available for gfx950 currently
+  info->useWarpSpeed = (info->algorithm == NCCL_ALGO_RING); // Enabled by default for any RING algorithm when platform supports it
+  if(rcclParamWarpSpeedAutoMode() != 0) {
```

Comment:
if rcclParamWarpSpeedAutoMode() is zero,   info->useWarpSpeed will be always true with ring? There is no else handling